hoomd/hpmc/external/wall.py

```diff
@@ -1,525 +1,152 @@
-        the ratio of the volume of the global box to the volume of the box that the walls class is currently storing. (After the scaling the box that the walls class is currently storing is updated appropriately.)
-        If you want to change the simulation box WITHOUT scaling the walls, then, you must first update the simulation box that the walls class is storing, THEN update the global box associated with the system definition.
+    Example::
 
-        Example::
+        mc = hoomd.hpmc.integrate.Sphere()
+        walls = [hoomd.wall.Sphere(radius=4.0)]
+        wall_potential = hoomd.hpmc.external.wall.WallPotential(walls)
+        mc.external_potential = wall_potential
 
-            init_box = hoomd.data.boxdim(L=10, dimensions=3);
-            snap = hoomd.data.make_snapshot(N=1, box=init_box, particle_types=['A']);
-            system = hoomd.init.read_snapshot(snap);
-            system.particles[0].position = [0,0,0];
-            system.particles[0].type = 'A';
-            mc = hpmc.integrate.sphere(seed = 415236);
-            mc.shape_param.set('A', diameter = 2.0);
-            ext_wall = hpmc.compute.wall(mc);
-            ext_wall.add_sphere_wall(radius = 3.0, origin = [0, 0, 0], inside = True);
-            ext_wall.set_curr_box(Lx=2.0*init_box.Lx, Ly=2.0*init_box.Ly, Lz=2.0*init_box.Lz, xy=init_box.xy, xz=init_box.xz, yz=init_box.yz);
-            system.sysdef.getParticleData().setGlobalBox(ext_wall.get_curr_box()._getBoxDim())
+    """
 
-        """
-        # much of this is from hoomd's update.py box_resize class
-        if all((Lx is None, Ly is None, Lz is None, xy is None, xz is None,
-                yz is None)):
-            hoomd.context.current.device.cpp_msg.warning(
-                "compute.wall.set_curr_box: Ignoring request to set the wall's \
-                        box without parameters\n")
+    def __init__(self, walls):
+        self._walls = hoomd.wall._WallsMetaList(walls, _to_hpmc_cpp_wall)
+
+    def _check_valid_shape_wall_pairs(self, walls, integrator):
+        supported_shape_wall_overlap_checks = {
+            hoomd.hpmc.integrate.Sphere: [
+                hoomd.wall.Sphere, hoomd.wall.Cylinder, hoomd.wall.Plane
+            ],
+            hoomd.hpmc.integrate.ConvexPolyhedron: [
+                hoomd.wall.Sphere, hoomd.wall.Cylinder, hoomd.wall.Plane
+            ],
+            hoomd.hpmc.integrate.ConvexSpheropolyhedron: [
+                hoomd.wall.Sphere, hoomd.wall.Plane
+            ]
+        }
+        integrator_type = type(integrator)
+        for wt in (type(w) for w in walls):
+            if wt not in supported_shape_wall_overlap_checks.get(
+                    integrator_type, []):
+                msg = f'Overlap checks between {wt} and {integrator_type} are '
+                msg += 'not supported.'
+                raise NotImplementedError(msg)
+
+    def _attach(self):
```

Comment:
Do you mean something different than what I implemented with `_check_valid_shape_wall_pairs` here?

Reply:
If it is too much trouble, we don't need to check for this error. We do document what is supported and what is not.

Reply:
I'm a little confused about the original comment. Is `_check_valid_shape_wall_pairs` defined [here](https://github.com/glotzerlab/hoomd-blue/blob/feature/hpmc-walls/hoomd/hpmc/external/wall.py#L91) and called [a few lines below this comment](https://github.com/glotzerlab/hoomd-blue/blob/feature/hpmc-walls/hoomd/hpmc/external/wall.py#L118) and [in the setter](https://github.com/glotzerlab/hoomd-blue/blob/feature/hpmc-walls/hoomd/hpmc/external/wall.py#L145) not dealing with the case where a user adds a new wall that is incompatible?

While we do document what is supported and what is not, I think we should raise this error since this seems like it could be a fairly easy gotcha since 1) wall support is not all-or-nothing for all shapes (e.g., `ConvexSpheropolyhedra` support spherical and planar but not cylindrical walls) and 2) the visibility in the docs is pretty low, just because of how much information is (necessarily) on that page.

Reply:
Ah, I see what you mean. Yes, I will need help with that.